Automattic/newspack-plugin
# Memberships Landing Page Wizard

### All Submissions:






### Changes proposed in this Pull Request:



Closes #168  .

This wizard implements a wizard that creates a memberships landing page pre-populated to feature available subscriptions. It also an optional `showOnBlockEditor` prop to our `Handoff` component that will also render the handoff in the block editor if enabled.

I think the flow I've made is very straightforward, but if you can think of ways it could be improved, please let me know. One thing we may want is for the wizard to redirect to editing the Memberships page when the Create button is clicked; I didn't implement that in this version because I think the current way provides valuable information about how to proceed, and that information would be lost if we just redirected. I'm open to implementing it either way if there is a strong preference.

I'm also open to feedback about improvements to the auto-generated content. Currently it's an intro paragraph and a featured product block for each subscription. The idea is that the user would add to the page to make it their own before publishing it.

Screenshots in details:
<details>

![localhost_wp-admin_admin php_page=newspack-memberships-page-wizard](https://user-images.githubusercontent.com/7317227/60915327-72414e80-a240-11e9-94d0-34bd32e2f920.png)
![localhost_wp-admin_admin php_page=newspack-memberships-page-wizard (1)](https://user-images.githubusercontent.com/7317227/60915331-73727b80-a240-11e9-806e-40a0493fcd07.png)
![localhost_wp-admin_post php_post=1800 action=edit](https://user-images.githubusercontent.com/7317227/60915337-753c3f00-a240-11e9-89e1-97038a5eef93.png)
![localhost_wp-admin_admin php_page=newspack-memberships-page-wizard (2)](https://user-images.githubusercontent.com/7317227/60915341-779e9900-a240-11e9-997c-edec5b70032f.png)

</details>

### How to test the changes in this Pull Request:

0. Make sure you have at least 1 subscription product created in the Subscriptions wizard or Donations set up.
1. Go to "Customize your memberships page" in the Reader Revenue checklist.
2. Click the button to Create the memberships page.
3. Click the button to Edit the memberships page.
4. Edit and customize the memberships page. Publish it. Optional: View it on the frontend.
5. Click the notice to return to the wizard. when finished.

### Other information:
